setup.sh

```diff
@@ -101,6 +48,73 @@ fatal() {
     exit 1
 }
 
+is_installed() {
+    name=$1
+    which "$name" > /dev/null 2>&1
+}
+
+# version of chalk to download
+version=${CHALK_VERSION:-latest}
+# which config to load after install
+load=${CHALK_LOAD:-}
+# json params to load
+params=${CHALK_PARAMS:-}
+# whether to automatically determine token via openid connect
+connect=${CHALK_CONNECT:-}
+# name of the custom profile to load
+profile=${CHALK_PROFILE:-default}
+# CrashOverride API token
+token=${CHALK_TOKEN:-}
+# OIDC token used to retrieve chalk token
+oidc=${CHALK_OIDC:-}
+# ${prefix}/bin is where script should install chalk and wrapped commands
+prefix=${CHALK_PREFIX:-/usr/local}
+# whether to overwrite existing chalk binary
+overwrite=${CHALK_OVERWRITE:-true}
+# whether to wrap external commands with chalk
+wrap=${CHALK_WRAP:-true}
+# chalk commands log level
+log_level=${CHALK_LOG_LEVEL:-error}
+# if running in debug mode
+debug=${CHALK_DEBUG:-}
+# instead of downloading chalk, copy it from this path
+# this is meant for testing local chalk binaries
+copy_from=${CHALK_COPY_FROM:-}
+# chalk command timeout
+timeout=${CHALK_TIMEOUT:-60}
+# which platforms to download for multi-platform builds
+platforms=${CHALK_PLATFORMS:-}
+# information for signing
+password=${CHALK_PASSWORD:-}
+public_key=${CHALK_PUBLIC_KEY:-}
+private_key=${CHALK_PRIVATE_KEY:-}
+# run chalk setup
+setup=${CHALK_SETUP:-}
+
+URL_PREFIX=https://dl.crashoverride.run
+SHA256=sha256sum
+SUDO=sudo
+TMP=/tmp
+
+# on osx, sha256sum doesnt exist and instead its shasum
+if ! is_installed "$SHA256"; then
+    SHA256="shasum -a 256"
+fi
+
+if ! is_installed sudo; then
+    SUDO=
+fi
+
+PROFILE=https://chalk.crashoverride.run/v0.1/profile
+GITHUB_OPENID_CONNECT=https://chalk.crashoverride.run/v0.1/openid-connect/github
+GITLAB_OPENID_CONNECT=https://chalk.crashoverride.run/v0.1/openid-connect/gitlab
+if [ -n "${__CHALK_TESTING__:-}" ]; then
```

Comment:
Non-blocking: We should document this feature clearly in notion before we forget and call it out on a team meeting